serve/mlc_serve/engine/engine_common.py

```diff
@@ -229,26 +229,46 @@ def get_requests_to_process(
     return requests, is_prompt_batch, token_counts
 
 
-def should_stop_by_length(state: RequestState, max_context_length: int) -> bool:
+def should_stop_seq_by_length(
+    gen_seq: GenerationSequence,
+    prompt_len: int,
+    max_context_length: int,
+    max_tokens: Optional[int],
+) -> bool:
     # TODO: currently, we simply return true for both stopping reasons.
     #       in the future, we can differentiate these two.
     # this include prompt tokens and gen tokens so far
-    if state.is_finished or state.stopping_criteria.max_tokens is None:
+
+    # If max_tokens is None, we do not put any length restriction.
+    if gen_seq.is_finished or max_tokens is None:
         return False
 
-    for gen_seq in state.generation_sequences:
-        if gen_seq.is_finished:
-            continue
+    num_context_tokens = prompt_len + len(gen_seq.generated_token_ids)
```

Comment:
Seems like I missed it when we discussed this over slack last time. 

I still think hard stop on `max_context_length` is an operational decision rather than the engine limitation, so would like to change this to something more related to engine configs, such as `max_input_len` or `max_num_batched_token`. Wonder what you guys think.

Reply:
> Seems like I missed it

What is "it"?

Reply:
the fact the engine still hard-stops at the max context length.

Reply:
I think we need to follow what other providers do, if there is a standard convention.